flaml/automl/task/time_series_task.py

```diff
@@ -0,0 +1,508 @@
+        return X
+
+    def preprocess(self, X, transformer=None):
+        if isinstance(X, pd.DataFrame) or isinstance(X, np.ndarray) or isinstance(X, pd.Series):
+            X = X.copy()
+            X = normalize_ts_data(X, self.target_names, self.time_col)
+            return self._preprocess(X, transformer)
+        elif isinstance(X, int):
+            return X
+        else:
+            raise ValueError(f"unknown type of X, {X.__class__}")
+
+    def evaluate_model_CV(
+        self,
+        config,
+        estimator,
+        X_train_all,
+        y_train_all,
+        budget,
+        kf,
+        eval_metric,
+        best_val_loss,
+        cv_score_agg_func=None,
+        log_training_metric=False,
+        fit_kwargs={},
+        free_mem_ratio=0,  # what is this for?
+    ):
+        if cv_score_agg_func is None:
+            cv_score_agg_func = default_cv_score_agg_func
+        start_time = time.time()
+        val_loss_folds = []
+        log_metric_folds = []
+        metric = None
+        train_time = pred_time = 0
+        total_fold_num = 0
+        n = kf.get_n_splits()
+        if self.is_classification():
+            labels = np.unique(y_train_all)
+        else:
+            labels = fit_kwargs.get("label_list")  # pass the label list on to compute the evaluation metric
+        ts_data = X_train_all
+        budget_per_train = budget / n
+        ts_data = X_train_all
+        for data in ts_data.cv_train_val_sets(kf.n_splits, kf.test_size, kf.step_size):
+            estimator.cleanup()
+            val_loss_i, metric_i, train_time_i, pred_time_i = get_val_loss(
+                config,
+                estimator,
+                X_train=data,
+                y_train=None,
+                X_val=data,
+                y_val=None,
+                eval_metric=eval_metric,
+                labels=labels,
+                budget=budget_per_train,
+                log_training_metric=log_training_metric,
+                fit_kwargs=fit_kwargs,
+                task=self,
+                weight_val=None,
+                groups_val=None,
+                free_mem_ratio=free_mem_ratio,
+            )
+            if isinstance(metric_i, dict) and "intermediate_results" in metric_i:
+                del metric_i["intermediate_results"]
+            total_fold_num += 1
+            val_loss_folds.append(val_loss_i)
+            log_metric_folds.append(metric_i)
+            train_time += train_time_i
+            pred_time += pred_time_i
+            if time.time() - start_time >= budget:
+                break
+        val_loss, metric = cv_score_agg_func(val_loss_folds, log_metric_folds)
+        n = total_fold_num
+        pred_time /= n
+        return val_loss, metric, train_time, pred_time
+
+    def default_estimator_list(self, estimator_list: List[str], is_spark_dataframe: bool) -> List[str]:
+        assert not is_spark_dataframe, "Spark is not yet supported for time series"
```

Comment:
@thinkall is this true in previous version?

Reply:
Currently in the case of spark dataframes, we only allow [estimators that are specifically for spark](https://github.com/microsoft/FLAML/blob/9977a7aae1d5d2bb7e6c09c23ec93802a95f6bd8/flaml/automl/task/generic_task.py#L958), and currently none of these do time series (in fact I believe the only one there right now is `lgbm_spark: SparkLGBMEstimator`

Reply:
Thanks. In that case, let's say "Spark dataframe is not ..."

Reply:
@EgorKraevTransferwise @ZmeiGorynych seems like this was missed. please make this change.

Reply:
What change? Spark is not supported yet for TS and we say so in the assert.